zaza/openstack/charm_tests/glance/tests.py

```diff
@@ -222,3 +223,46 @@ def test_100_create_delete_image(self):
         )
         self.assertEqual(image["size"], response["ContentLength"])
         openstack_utils.delete_image(self.glance_client, image["id"])
+
+
+class GlanceCinderBackendTest(test_utils.OpenStackBaseTest):
+    """Encapsulate glance tests using cinder backend."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running glance tests with cinder backend."""
+        super(GlanceCinderBackendTest, cls).setUpClass()
+        cls.glance_client = openstack_utils.get_glance_session_client(
+            cls.keystone_session)
+        cls.cinder_client = openstack_utils.get_cinder_session_client(
+            cls.keystone_session)
+
+    def test_100_create_delete_image(self):
+        """Create an image and do a simple validation of it.
+
+        Validate the size of the image in both Glance API and Cinder API.
+        """
+        image_name = "zaza-cinder-test-image"
+        openstack_utils.create_image(
+            glance=self.glance_client,
+            image_url=openstack_utils.find_cirros_image(arch="x86_64"),
+            image_name=image_name,
+            backend="cinder",
+        )
+        images = openstack_utils.get_images_by_name(
+            self.glance_client, image_name)
+        self.assertEqual(len(images), 1)
+        image = images[0]
+
+        volume_name = 'image-'+image["id"]
+        volumes = openstack_utils.get_volumes_by_name(
+            self.cinder_client, volume_name)
+        self.assertEqual(len(volumes), 1)
+        volume = volumes[0]
+
+        logging.info(
+            "Checking glance image size {} matches volume size {} "
+            "GB".format(image["size"], volume.size))
+        image_size_in_gb = int(math.ceil(float(image["size"]) / 1024 ** 3))
+        self.assertEqual(image_size_in_gb, volume.size)
+        openstack_utils.delete_image(self.glance_client, image["id"])
```

Comment:
Is this async in the cloud?  Worth checking if it is, and whether the test should wait here until the image is deleted so that it doesn't have the potential to pollute other tests.  If no async in the cloud, then please ignore this comment.

Reply:
DELETE API call returns 204 and so delete image is synchronous. Ignoring the comment

Reply:
> DELETE API call returns 204 and so delete image is synchronous. Ignoring the comment

Just to be slightly pedantic, I don't think this is true, but tracing the `openstack_utils.delete_image()` eventually calls `resource_removed()` which does wait for the resource to be deleted, so `delete_image()` is forced to be sync.

I think HTTP code 204 just means that the service accepted the request and there is no content.  e.g. it is a 200 with no content.  The semantics of the request, being sync or async, isn't embodied in the return code.  Happy to be corrected, though!